src/amo/imageUtils.js

```diff
@@ -1,7 +1,24 @@
 import fallbackIcon from 'amo/img/icons/default.svg';
 
-export function getAddonIconUrl(addon) {
-  return addon ? addon.icon_url : fallbackIcon;
+// returns icon url from icons objects,
+// or returns icon_url value if icons does't exists
+//
+// fallback icon will be returned if the addon has no icon.
+//
+// iconSize values are 32 | 64 | 128
+export function getAddonIconUrl(addon, iconSize) {
+  const defaultIconSize = 64;
+
+  const addonIconsExists = addon?.icons;
+
+  if (addonIconsExists) {
+    const iconFromIcons = addon.icons?.[iconSize || defaultIconSize];
+
+    return iconFromIcons || fallbackIcon;
```

Comment:
what about when `iconSize` is defined but doesn't exist on the object (or the edge case when `iconSize` isn't defined, so  defaultIconSize is used, but doesn't exist - `AddonIconsType` you've defined allows `64` not being defined)?

Reply:
Handled and there is a test for this case now.